include/zephyr/drivers/comparator.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_COMPARATOR_H_
+#define ZEPHYR_INCLUDE_DRIVERS_COMPARATOR_H_
+
+/**
+ * @brief Comparator Interface
+ * @defgroup comparator_interface Comparator Interface
+ * @since 3.7
+ * @version 0.1.0
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/device.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Comparator trigger enumerations */
+enum comparator_trigger {
+	/** No trigger */
+	COMPARATOR_TRIGGER_NONE = 0,
+	/** Trigger on rising edge of comparator output */
+	COMPARATOR_TRIGGER_RISING_EDGE,
+	/** Trigger on falling edge of comparator output */
+	COMPARATOR_TRIGGER_FALLING_EDGE,
+	/** Trigger on both edges of comparator output */
+	COMPARATOR_TRIGGER_BOTH_EDGES
+};
+
+/** Comparator callback template */
+typedef void (*comparator_callback_t)(const struct device *dev, void *user_data);
+
+/** @cond INTERNAL_HIDDEN */
+
+typedef int (*comparator_api_get_output)(const struct device *dev);
```

Comment:
should this be just "get"? like `gpio_pin_get`

Reply:
The reason I'm specifying output here is to not confuse it with the state of the inputs :) The output could be inverted for example (high if positive input is lower than negative input), so `_output` here hopefully prevents confusing the logical state with the output state :)